pkg/util/phone/legal_parser.go

```diff
@@ -0,0 +1,68 @@
+package phone
+
+import (
+	"regexp"
+	"strconv"
+
+	"github.com/nyaruka/phonenumbers"
+)
+
+// Login ID normalizer uses LegalParser
+// What is Legal? phonenumbers.Parse does not return err.
+type LegalParser struct{}
```

Comment:
I think we should instantiate instance of LegalParser and LegalAndValidParser, so when using them we don't need to create new instance.

And also, there should be an interface `Parser` which is implemented by LegalParser and LegalAndValidParser.

```golang
var LegalParser Parser = legalParser{}
var LegalAndValidParser Parser = legalAndValidParser{}
```

Reply:
Also, no need to put LegalParser and LegalAndValidParser in the wire dependency graph. They are just utilities we can use directly without dependency injection.

Reply:
Please tidy up the comment.

The doc of struct in Go should start with the struct name. So the doc should read

> LegalParser parses a legal phone number. A legal phone number is a phone number that passes phonenumbers.Parse().